loki/transformations/single_column/vector.py

```diff
@@ -450,6 +480,196 @@ def transform_subroutine(self, routine, **kwargs):
+            routine.body = self.remove_vector_sections(routine.body)
+
+            with pragmas_attached(routine, ir.Loop):
+                # Check for explicitly labelled vector-reduction regions
+                self.mark_vector_reductions(routine, routine.body)
+
+                # Mark sequential loops inside vector sections
+                self.mark_seq_loops(routine.body)
+
+            # Mark subroutine as vector parallel for later annotation
+            # routine.spec.append(ir.Pragma(keyword='loki', content='routine vector'))
+            routine.spec.append(ir.Pragma(keyword='loki', content='routine seq'))
+
+        if role == 'driver':
+
+            # add horizontal.index, e.g., 'jl'
+            index = get_integer_variable(routine, self.horizontal.index)
+            routine.variables += (index,)
+
+            with pragmas_attached(routine, ir.Loop):
+                driver_loops = find_driver_loops(section=routine.body, targets=targets)
+
+                for loop in driver_loops:
+
+                    # Wrap calls being in targets in a horizontal loop and add horizontal.index as argument
+                    call_map = {}
+                    for call in FindNodes(ir.CallStatement).visit(loop.body):
+                        if str(call.name).lower() in targets:
+                            if self.horizontal.index not in call.arg_map:
+                                new_kwarg = (self.horizontal.index,
+                                        get_integer_variable(routine, self.horizontal.index))
+                                updated_call = call.clone(kwarguments=call.kwarguments + (new_kwarg,))
+                                call_arg_map = {k.name.lower(): v for (k, v) in call.arg_map.items()}
+                                # loop bound(s) could be derived types ...
+                                ltmp = self._get_loop_bound(self.horizontal.lower, call_arg_map)
+                                utmp = self._get_loop_bound(self.horizontal.upper, call_arg_map)
+                                lower = ltmp[0] if ltmp[1] is None else sym.Variable(name=ltmp[1], parent=ltmp[0])
+                                upper = utmp[0] if utmp[1] is None else sym.Variable(name=utmp[1], parent=utmp[0])
+                                # wrap call with horizontal loop
+                                loop_bounds = (lower, upper)
+                                call_map[call] = wrap_vector_section((updated_call,), routine, bounds=loop_bounds,
+                                        insert_pragma=True, index=self.horizontal.index)
+
+                    loop._update(body=Transformer(call_map).visit(loop.body))
+
+                    # Revector all marked sections within the driver loop body
+                    loop._update(body=self.revector_section(routine, loop.body))
+
+                    # Check for explicitly labelled vector-reduction regions
+                    super().mark_vector_reductions(routine, loop.body)
```

Comment:
[no action] Ahh, so in driver compute regions we fall back to SCCVec behaviour, yes? Very neat!

Reply:
Exactly, yes.